plugins/modules/device_credential_intent.py

```diff
@@ -0,0 +1,2323 @@
+                    flag = True
+                    credential_params.update({value: want_update.get(value)[i]})
+            i = i + 1
+            if credential_params:
+                final_response.append(credential_params)
+                response = self.dnac._exec(
+                    family="discovery",
+                    function='update_global_credentials_v2',
+                    params=credential_params,
+                )
+                self.log(str(response))
+                if not response:
+                    self.msg = "response is empty"
+                    self.status = "exited"
+                    return self
+
+                if not isinstance(response, dict):
+                    self.msg = "response is not a dictionary"
+                    self.status = "exited"
+                    return self
+
+                task_id = response.get("response").get("taskId")
+                while True:
+                    task_details = self.get_task_details(task_id)
+                    self.log(str(task_details))
+                    if task_details.get("isError") is True:
+                        self.msg = str(task_details.get("progress"))
+                        self.status = "failed"
+                        return self
+                    elif task_details.get("isError") is False:
+                        self.result['changed'] = True
+                        break
+        self.log("Update Device Credential API input - " + str(final_response))
+        self.log("Global Device Credential Updated Successfully")
+        result_global_credential.update({
+            "Updation": {
+                "response": final_response,
+                "msg": "Global Device Credential Updated Successfully"
+            }
+        })
+        self.msg = "Global Device Credential Updated Successfully"
+        self.status = "success"
+        return self
+
+    def assign_credentials_to_site(self):
+        """
+        Assign Global Device Credential to the Cisco DNA
+        Center based on the provided playbook details.
+        Check the return value of the API with check_return_status().
+
+        Parameters:
+            None
+
+        Returns:
+            self
+        """
+
+        result_assign_credential = self.result.get("response")[0].get("assignCredential")
+        credential_params = self.want.get("assign_credentials")
+        final_response = []
+        self.log("Assign Device Credential to site API input - " + str(credential_params))
+        if not credential_params:
+            result_assign_credential.update({
+                "No Assign Credentials": {
+                    "response": "No Response",
+                    "msg": "No Assignment is available"
+                }
+            })
+            self.msg = "No Assignment is available"
+            self.status = "success"
+            return self
+        for site_id in self.want.get("site_id"):
```

Comment:
instead of getting siteids everytime, we can do this before this for loop..
```
    site_ids = self.want.get("site_id")

    for site_id in site_ids:
```